Kernel/Arch/x86_64/ISABus/I8042Controller.h

```diff
@@ -66,97 +65,88 @@ enum I8042Response : u8 {
     {
         SpinlockLocker lock(m_lock);
-        return do_send_command(device, command);
-    }
-    ErrorOr<u8> send_command(PS2Device::Type device, u8 command, u8 data)
-    {
-        SpinlockLocker lock(m_lock);
-        return do_send_command(device, command, data);
+        return do_read_from_device(port_index);
     }
+    virtual ErrorOr<void> prepare_for_input(PS2PortIndex) override;
+    virtual bool irq_process_input_buffer(PS2PortIndex) override;
 
-    ErrorOr<u8> read_from_device(PS2Device::Type device)
-    {
-        SpinlockLocker lock(m_lock);
-        return do_read_from_device(device);
-    }
+    // Note: This function exists only for the initialization process of the controller
+    bool check_existence_via_probing(Badge<HIDManagement>);
+
+    bool handle_irq(Badge<I8042ControllerIRQHandler>, u8 irq_number);
 
-    ErrorOr<void> wait_then_write(u8 port, u8 data)
+private:
+    I8042Controller();
+
+    ErrorOr<void> send_command_to_specific_port(PS2PortIndex port_index, u8 command)
     {
         SpinlockLocker lock(m_lock);
-        return do_wait_then_write(port, data);
+        TRY(do_send_command(port_index, command));
+        return {};
     }
-
-    ErrorOr<u8> wait_then_read(u8 port)
+    ErrorOr<void> send_command_to_specific_port(PS2PortIndex port_index, u8 command, u8 data)
     {
         SpinlockLocker lock(m_lock);
-        return do_wait_then_read(port);
+        TRY(do_send_command(port_index, command, data));
+        return {};
     }
 
-    ErrorOr<void> prepare_for_output();
-    ErrorOr<void> prepare_for_input(PS2Device::Type);
+    ErrorOr<void> prepare_for_any_output();
 
-    bool irq_process_input_buffer(PS2Device::Type);
+    ErrorOr<void> prepare_for_any_input();
 
-    // Note: This function exists only for the initialization process of the controller
-    bool check_existence_via_probing(Badge<HIDManagement>);
-
-private:
-    I8042Controller();
-    ErrorOr<void> do_reset_device(PS2Device::Type);
-    ErrorOr<u8> do_send_command(PS2Device::Type type, u8 data);
-    ErrorOr<u8> do_send_command(PS2Device::Type device, u8 command, u8 data);
-    ErrorOr<u8> do_write_to_device(PS2Device::Type device, u8 data);
-    ErrorOr<u8> do_read_from_device(PS2Device::Type device);
+    ErrorOr<void> do_reset_device(PS2PortIndex);
+    ErrorOr<void> do_send_command(PS2PortIndex port_index, u8 data);
+    ErrorOr<void> do_send_command(PS2PortIndex port_index, u8 command, u8 data);
+    ErrorOr<void> do_write_to_device(PS2PortIndex port_index, u8 data);
+    ErrorOr<u8> do_read_from_device(PS2PortIndex port_index);
     ErrorOr<void> do_wait_then_write(u8 port, u8 data);
-    ErrorOr<u8> do_wait_then_read(u8 port);
+
+    // NOTE: The meaning of "any input" here is that this is not attached
+    // to any PS2 port, but rather we accept any serial input, which is vital
+    // when reading values before initializing any actual PS2 device!
+    ErrorOr<u8> do_wait_then_read_any_input(u8 port);
```

Comment:
i appreciate the comment and i think i understand it, but as i tend to overstress the naming, iʼd pick on this and try to challenge that. 😅

iʼd avoid “any” here as not specifying with a qualifier would mean we care for all/any input. if youʼd go for something specific, youʼd mention it here. in other cases this stress in form of “any” is making you think of what other special types of input you would have.

in such case iʼd go for other alignments too, if applicable (it seems there would be some, according to the comment), e.g. “do wait then read device input”? 🤔

or maybe youʼd like to stress something about controller communication instead?

Reply:
I think this is fine as it is. This is not about reading input from a device - it could be generated from anywhere (device or controller too).